### PR TITLE
MAT- 6901 fix HQMF issues 

### PR DESCRIPTION
**Pull request by adongare**
## MADiE PR

Jira Ticket: [MAT-6901](https://jira.cms.gov/browse/MAT-6901)
(Optional) Related Tickets:
related pr: https://github.com/MeasureAuthoringTool/cql-elm-translation/pull/160

### Summary
This PR contains fixes for following issues:
- referential integrity among UUIDs referred 
- Measure References contain spaces.  e.g. `<value mediaType="text/plain" value="Citation                test 2" xsi:type="ED"/> `
- missing precondition for criteriaReferences of a populationCriteriaSection components. rach population criteria refers to the UUID of root of the library for which it belongs to 
- Correcting URLs: all emasuretool urls should be replaced with madie.cms.gov
- missing observation for ratio & CV - done. -done
- functions(may have some issues, haven't tested yet)
- missing populationCriteriaSection components for RAVs, STRAT  

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/services/MeasureMapper.java [hunk 1]
```diff
@@ -142,82 +152,135 @@ default MeasureGroupingType measureToMeasureGroupingType(QdmMeasure measure) {
...
+                  })
+              .toList());
     }
     return CollectionUtils.isEmpty(clauses) ? null : clauses;
   }
 
+  default CQLDefinition getCqlDefinition(String definition, Set<CQLDefinition> cqlDefinitions) {
+    return cqlDefinitions.stream()
+        .filter(cqlDefinition -> definition.equals(cqlDefinition.getDefinitionName()))
+        .findFirst()
```
Comment (nmorasb) [inline src/main/java/gov/cms/madie/services/MeasureMapper.java:225]: we'll have to keep this in mind when we re-enable Ratio w/ 2 IPs..

**Review by nmorasb — APPROVED** [review]

File: src/main/java/gov/cms/madie/services/MeasureMapper.java (hunk 1 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/services/MeasureMapper.java:225]: yeah, right

Comment (adongare) [reply src/main/java/gov/cms/madie/services/MeasureMapper.java:225]: or maybe not...I think it would be fine. CQL definitions are unique.
